src/data/dlist/basic.lean

```diff
@@ -10,3 +10,6 @@ difference list.  Similar to `list.join`. -/
 def dlist.join {α : Type*} : list (dlist α) → dlist α
  | [] := dlist.empty
  | (x :: xs) := x ++ dlist.join xs
+
+@[simp] lemma dlist_singleton {α : Type*} {a : α} :
```

Comment:
I don't know anything about the simp normal form for `dlist`, is there a reason to prefer the RHS? Are there other simp lemmas about `lazy_of_list`?

Reply:
I'm working on making lemmas about parsers. Since `decorate_errors` places errors into `lazy_of_list` like so:
```
def decorate_errors (msgs : thunk (list string)) (p : parser α) : parser α :=
λ input pos, match p input pos with
| parse_result.fail _ expected :=
  parse_result.fail pos (dlist.lazy_of_list (msgs ()))
| ok := ok
```

then having the simp lemma in the PR allows a lemma like this:
```
lemma ch_of_empty (c : char) (cb : char_buffer) (h : cb.size = 0) (n : ℕ) :
  ch c cb n = parse_result.fail n (dlist.singleton c.to_string) :=
by simp [ch, sat, h]
```

Reply:
`dlist` is isomorphic to `list`. I think a reasonable normal form for `dlist` is `dlist.of_list <list...>` because we are sure to have more theorems about the latter.

Reply:
So have an additional lemma like:
```
@[simp] lemma dlist_lazy {α : Type*} {l : list α} :
  dlist.lazy_of_list l = dlist.of_list l := rfl
```
?

Reply:
If that lemma is true, it seems like a good idea; better to limit the number of functions we have to deal with in proofs.